infra/resources/_modules/monitoring/action_group.tf

```
@@ -0,0 +1,11 @@
+resource "azurerm_monitor_action_group" "io_com_error_action_group" {
```

The `_action_group` suffix is unnecessary, because this resource will always be addressed by its [address](https://developer.hashicorp.com/terraform/cli/state/resource-addressing#resource-spec) (`azurerm_monitor_action_group.io_com_error`)

```suggestion
resource "azurerm_monitor_action_group" "io_com_error" {
```